### PR TITLE
Change base image to Debian Bookworm

### PR DESCRIPTION
**Pull request by vfusco**
and some small updates and fixes

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -5,45 +5,77 @@ env:
...
       - name: Download linux
         run: make download
 
-      - name: Build and push docker image
-        id: docker_build
-        uses: docker/build-push-action@v2
+      - name: Set up Depot CLI
+        uses: depot/setup-action@v1
+
+      - name: Build and push docker image to GitHub Container Registry
```
Comment (tuler) [inline .github/workflows/build.yml:54]: You don't need a separate step to publish to GHCR.
It's better to control in the meta step using `enable`

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -5,45 +5,84 @@ env:
...
-        uses: docker/metadata-action@v3
+        uses: docker/metadata-action@v4
+        with:
+          images: docker.io/${{ github.repository_owner }}/toolchain
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{version}}
+
+      - name: GitHub Registry meta
+        id: ghcr_meta
```
Comment (tuler) [inline .github/workflows/build.yml:35]: Not there yet 🙂
Use a single meta
Just add a condition like [this](https://github.com/cartesi/rollups/blob/40aa6af3ab7668c698e9b3eac03621b3c585c11f/.github/workflows/docker.yml#L39)

File: .github/workflows/build.yml (hunk 1 above)
Comment (tuler) [reply .github/workflows/build.yml:61]: It tried to publish on docker hub because the meta is wrong

**Review by tuler — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -11,9 +11,9 @@
 # the License.
 #
 
-FROM ubuntu:22.04
+FROM debian:bookworm-20230612 as ct-ng-builder
```
Comment (mpolitzer) [inline Dockerfile:14]: Why so specific and not `debian:bookworm`?

**Review by mpolitzer — APPROVED** [review]

**Pull request merged**